perfscale_regression_ci/scripts/network/network_policy_scalability.sh

```diff
@@ -0,0 +1,148 @@
+
+pass_fail=0
+workload_template_list=("${DIR}/../../kubeburner-object-templates/scaling-network-policy-config.yaml" "${DIR}/../../kubeburner-object-templates/scaling-no-network-policy-config.yaml")
+pod_scale_max=2000
+kube_burner_job_list=("np-issue-test" "no-np-issue-test")
+final_time_list=()
+deny_network_policy="${DIR}/../../content/deny_network_traffic_policy.yaml"
+deny_traffic_code=000
+return_traffic_code=200
+
+
+echo "======Use kube-burner to load the cluster with test objects======"
+
+for ((i = 0; i < 2; i++));
+do
+    export WORKLOAD_TEMPLATE=${workload_template_list[$i]}
+    export NAME=${kube_burner_job_list[$i]}
+    export NAMESPACE=${kube_burner_job_list[$i]}
+    deployment_name=${kube_burner_job_list[$i]}
+    scale_up=0
+
+    echo "Starting scale-up test for ${kube_burner_job_list[$i]}..."
+    run_workload
+    oc scale --replicas $pod_scale_max deployment $deployment_name -n ${NAMESPACE}; start_time=`date +%s`
+    echo "start time: $start_time"
+    scale_up=$(oc get deployment $deployment_name --no-headers -n ${NAMESPACE} | awk -F ' {1,}' '{print $4}')
+    check_deployment_pod_scale ${NAMESPACE} $deployment_name $scale_up $pod_scale_max
+    end_time=`date +%s`
+    echo "end time: $end_time"
+    final_time=$((end_time - start_time))
+    echo "execution time for test ${kube_burner_job_list[$i]}: $final_time s."
+    final_time_list+=($final_time)
+	sleep 1
+    echo "=========================================================================="
+done
+
+final_time_np=${final_time_list[0]}
+final_time_no_np=${final_time_list[1]}
+
+sleep 5
+
+pod_name=$(oc get po -n ${NAMESPACE} --no-headers | head -n 1 | awk '{print $1}')
+pod_ip=$(oc get po $pod_name -n  ${NAMESPACE} --output jsonpath='{.status.podIP}')
+apiserver_pod=$(oc get po -n openshift-oauth-apiserver --no-headers | head -n 1 | awk '{print $1}')
+
+echo "Starting test to deny network traffic in the namespace ${NAMESPACE}..."
+oc apply -f $deny_network_policy -n ${NAMESPACE}
+deny_time_start=`date +%s`
+check_http_code $deny_traffic_code $pod_ip $apiserver_pod
+deny_time_end=`date +%s`
+deny_final_time=$((deny_time_end - deny_time_start))
+echo "Time taken for network policy to become inactive: $deny_final_time s"
+
+sleep 1
+
+echo "Starting test to allow network traffic in the namespace ${NAMESPACE}..."
+oc delete networkpolicy deny-by-default -n ${NAMESPACE}
+return_traffic_start=`date +%s`
+check_http_code $return_traffic_code $pod_ip $apiserver_pod
+return_traffic_end=`date +%s`
+return_traffic_final_time=$((return_traffic_end - return_traffic_start))
+echo "Time taken for network policy to become active: $return_traffic_final_time s"
+
+echo "=========================================================================="
+
+policy_no_policy_difference=$(calculate_difference ${final_time_np} ${final_time_no_np})
+deny_return_traffic_difference=$(calculate_difference ${deny_final_time} ${return_traffic_final_time})
+
+if [[ ( $final_time_np -le 120 ) ]]; then
```

Comment:
I think it would be nice to be able to move this line through 99 into a single if statement in your loop above. You could still print which scenario passed based on name of the test. This huge section of if statements is hard to tell what its actually verifying. Maybe we need to try to section off this test some so that we do the deployment scaling/load verification and then the deny traffic verification with its corresponding if statements.